### PR TITLE
Update OpenTelemetry documentation

### LINKED ISSUE #31921: [FEATURE REQ] Document ActivitySource names or naming conventions
### Library name

All

### Please describe the feature.

Users sometimes what to enable specific ActivitySources and exclude others.  We only document `Azure.*` and don't give any guidance on source names beyond that.

~~E.g. storage one is `Azure.Storage`,  appconfig one is `Azure.Data.AppConfiguration` and eventhubs is `Azure.Messaging.EventHubs`.~~

~~In some cases, they are defined explicitly (EventHubs), but in most cases, the name corresponds to the namespace in which derived `ClientOptions` are defined.~~

The convention: `<packageName>.<className>`.

E.g. ServiceBus should have the following sources

- `Azure.Messaging.ServiceBus.ServiceBusReceiver`
- `Azure.Messaging.ServiceBus.Message`
- `Azure.Messaging.ServiceBus.ServiceBusSender`
- `Azure.Messaging.ServiceBus.ServiceBusProcessor`

It's recommended to enable Azure SDK tracing in one of the following ways:
- `AddSource("Azure.*)` - all activities
- `AddSource("<packageName>.*)` 
  - all activities from specific library such as `Azure.Messaging.ServiceBus`
  - no activities for HTTP requests (they are coming from `Azure.Core.Http` and need to be enabled additionally)
- `AddSource("<packageName>.<clientName>")` - activities from individual clients (e.g. when `Azure.Messaging.ServiceBus.ServiceBusProcessor` is used, receiver activities won't be reported)

### PR DESCRIPTION
**Pull request by lmolkova**
Fixes #31921 and updates core docs to talk about OpenTelemetry rather than AppInsights

### CODE REVIEW

File: sdk/core/Azure.Core/samples/Diagnostics.md [hunk 1]
```diff
@@ -176,6 +179,117 @@ Most of Azure SDKs released in or after November 2023 have OpenTelemetry support
...
+
+Unlike generic HTTP activities, Azure SDK HTTP activities include Azure-specific attributes such as request identifiers usually passed to and from Azure serices in `x-ms-client-request-id`, `x-ms-request-id` or similar request and respose headers. This data may be important when correlating client and server telemetry or creating support tickets.
+
+To avoid double-collection you may either 
+- enrich generic HTTP client activities with Azure request identifiers and disable Azure SDK HTTP activities.
+- filter out duplicated generic HTTP client activities.
+
+#### Enriching generic HTTP activities with Azure request identifiers
+
+For example, you can do it with the following code snippet on .NET Core.
```
Comment (JoshLove-msft) [inline sdk/core/Azure.Core/samples/Diagnostics.md:221]: Do we still use .NET core as terminology?

**Review by JoshLove-msft — APPROVED** [review]

File: sdk/core/Azure.Core/README.md [hunk 2]
```diff
@@ -234,16 +234,14 @@ KeyVaultSecret secret = client.GetSecret("Name");
...
-
-If your application already uses ApplicationInsights, automatic collection of Azure SDK traces is supported since version `2.12.0`.
-
-To setup ApplicationInsights tracking for your application follow the [Start Monitoring Application](https://docs.microsoft.com/azure/azure-monitor/learn/dotnetcore-quick-start) guide.
+Azure SDKs are instrumented for distributed tracing using [OpenTelemetry](https://opentelemetry.io/). Distributed tracing allows to follow request through multiple services, record how long network or logical call take along with structured properties describing such operations.
 
 More on diagnostics in [diagnostics samples](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/samples/Diagnostics.md).
 
+To setup distributed tracing for your application follow your observability vendor documentation. If you use Azure Monitor, follow the [Start Monitoring Application](https://docs.microsoft.com/azure/azure-monitor/learn/dotnetcore-quick-start) guide.
```
Comment (vishweshbankwar) [inline sdk/core/Azure.Core/README.md:243]: Not introduced in this PR but looks like this link does not exist anymore?

File: sdk/core/Azure.Core/samples/Diagnostics.md [hunk 3]
```diff
@@ -162,10 +162,13 @@ using AzureEventSourceListener listener = new AzureEventSourceListener((args, me
 
 If your are using Azure SDK libraries in ASP.NET Core application consider using the `Microsoft.Extensions.Azure` package that provides integration with `Microsoft.Extensions.Logging` library. See [Microsoft.Extensions.Azure readme](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/extensions/Microsoft.Extensions.Azure/README.md) for more details.
 
-## ActivitySource support
+## Distributed tracing
+
+Azure SDKs are instrumented for distributed tracing using OpenTelemetry or ApplicationsInsights SDK.
```
Comment (vishweshbankwar) [inline sdk/core/Azure.Core/samples/Diagnostics.md:167]:
```suggestion
Azure SDKs are instrumented for distributed tracing using OpenTelemetry or ApplicationInsights SDK.
```

File: sdk/core/Azure.Core/samples/Diagnostics.md [hunk 4]
```diff
@@ -176,6 +179,117 @@ Most of Azure SDKs released in or after November 2023 have OpenTelemetry support
...
+
+#### Filtering out duplicated HTTP client activities
+
+Another approach would be to filter out generic HTTP client activities with the following approach:
+
+```csharp
+.WithTracing(tracerProviderBuilder => tracerProviderBuilder
+    .AddSource("Azure.*")
+    .AddHttpClientInstrumentation(o => {
+        o.EnrichWithHttpResponseMessage = (activity, response) =>
```
Comment (vishweshbankwar) [inline sdk/core/Azure.Core/samples/Diagnostics.md:259]: meant to show [this](https://github.com/Azure/azure-sdk-for-net/blob/cbc04f8981f3fca3dd18d905ff9549480f17c3b6/sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/src/OpenTelemetryBuilderExtensions.cs#L149-L164) ?

Comment (lmolkova) [reply sdk/core/Azure.Core/samples/Diagnostics.md:259]: great catch,, not sure how this has happened

File: sdk/core/Azure.Core/README.md [hunk 5]
```diff
@@ -234,16 +234,14 @@ KeyVaultSecret secret = client.GetSecret("Name");
 
 More on mocking in [Unit testing and mocking with the Azure SDK for .NET](https://learn.microsoft.com/dotnet/azure/sdk/unit-testing-mocking).
 
-## Distributed tracing with Application Insights
+## Distributed tracing with OpenTelemetry
```
Comment (rajkumar-rangaraj) [inline sdk/core/Azure.Core/README.md:237]: :+1:

**Review by annelo-msft — COMMENTED** [review]
❤️

**Pull request merged**